File: platforms/swift/Sources/ShopifyCheckoutKit/CheckoutEvent.swift
import Foundation

/// The payload delivered when checkout starts.
public struct CheckoutStartEvent: Equatable, Sendable {
    public let checkout: Checkout

    public init(checkout: Checkout) {
        self.checkout = checkout
    }
}

/// The payload delivered when the buyer-visible checkout state changes.
public struct CheckoutUpdateEvent: Equatable, Sendable {
    public let checkout: Checkout

    public init(checkout: Checkout) {
        self.checkout = checkout
    }
}

/// The payload delivered when checkout completes.
public struct CheckoutCompleteEvent: Equatable, Sendable {
    public let checkout: Checkout

    public init(checkout: Checkout) {
        self.checkout = checkout
    }
}

/// The payload delivered when checkout cannot continue.
public struct CheckoutFailureEvent {
    public let error: CheckoutError

    public init(error: CheckoutError) {
        self.error = error
    }
}

File: platforms/android/lib/src/main/java/com/shopify/checkoutkit/CheckoutEvent.kt
package com.shopify.checkoutkit

/** The payload delivered when checkout starts. */
public class CheckoutStartEvent public constructor(public val checkout: Checkout)

/** The payload delivered when the buyer-visible checkout state changes. */
public class CheckoutUpdateEvent public constructor(public val checkout: Checkout)

/** The payload delivered when checkout completes. */
public class CheckoutCompleteEvent public constructor(public val checkout: Checkout)

/** The payload delivered when checkout cannot continue. */
public class CheckoutFailureEvent public constructor(public val error: CheckoutException)
